### ISSUE TITLE
Core dump on Ubuntu if tests are run when MySQL service is stopped

### ISSUE THREAD

**Issue by blah238** [mscdex/node-mariasql#113]
```
vagrant@vagrant-ubuntu-trusty-64:~$ node --version
v4.2.1
vagrant@vagrant-ubuntu-trusty-64:~$ npm --version
2.14.8
vagrant@vagrant-ubuntu-trusty-64:~$ git clone https://github.com/mscdex/node-mariasql.git
Cloning into 'node-mariasql'...
remote: Counting objects: 2986, done.
remote: Compressing objects: 100% (4/4), done.
Receiving objects: 100% (2986/2986), 6.46 MiB | 3.77 MiB/s, done.
remote: Total 2986 (delta 0), reused 0 (delta 0), pack-reused 2982
Resolving deltas: 100% (1358/1358), done.
Checking connectivity... done.
vagrant@vagrant-ubuntu-trusty-64:~$ cd node-mariasql
vagrant@vagrant-ubuntu-trusty-64:~/node-mariasql$ npm install
npm WARN package.json mariasql@0.2.1 No license field.

> mariasql@0.2.1 install /home/vagrant/node-mariasql
> node-gyp rebuild

make: Entering directory `/home/vagrant/node-mariasql/build'
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/aes.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/aestables.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/algebra.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/arc4.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/asn.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/bftables.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/blowfish.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/coding.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/des.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/dh.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/dsa.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/file.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/hash.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/hc128.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/integer.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/md2.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/md4.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/md5.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/misc.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/rabbit.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/random.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/ripemd.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/rsa.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/sha.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/tftables.o
  CXX(target) Release/obj.target/taocrypt/deps/libmariadbclient/extra/yassl/taocrypt/src/twofish.o
  AR(target) Release/obj.target/deps/libmariadbclient/extra/yassl/taocrypt/taocrypt.a
  COPY Release/taocrypt.a
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/buffer.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/cert_wrapper.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/crypto_wrapper.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/handshake.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/lock.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/log.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/socket_wrapper.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/ssl.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/timer.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/yassl_error.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/yassl_imp.o
  CXX(target) Release/obj.target/yassl/deps/libmariadbclient/extra/yassl/src/yassl_int.o
  AR(target) Release/obj.target/deps/libmariadbclient/extra/yassl/yassl.a
  COPY Release/yassl.a
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/sql-common/client.o
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/sql-common/client_plugin.o
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/sql-common/mysql_async.o
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/sql-common/my_time.o
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/sql-common/pack.o
  CXX(target) Release/obj.target/clientlib/deps/libmariadbclient/sql/net_serv.o
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/sql/password.o
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/libmysql/errmsg.o
  CC(target) Release/obj.target/clientlib/deps/libmariadbclient/libmysql/libmysql.o
  AR(target) Release/obj.target/deps/libmariadbclient/libmysql/clientlib.a
  COPY Release/clientlib.a
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/array.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/charset-def.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/charset.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/errors.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/hash.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/list.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_arr_appstr.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_dirname.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_fn_ext.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_format.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_loadpath.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_pack.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_path.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_qsort.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_unixpath.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mf_wcomp.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/mulalloc.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_access.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_alloc.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_compress.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_context.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_default.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_div.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_error.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_fopen.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_fstream.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_getsystime.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_getwd.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_init.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_lib.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_malloc.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_mess.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_once.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_open.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_pthread.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_read.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_seek.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_static.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_symlink.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_thr_init.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_write.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/safemalloc.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/string.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/thr_mutex.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/typelib.o
  CC(target) Release/obj.target/mysys/deps/libmariadbclient/mysys/my_sync.o
  AR(target) Release/obj.target/deps/libmariadbclient/mysys/mysys.a
  COPY Release/mysys.a
  CXX(target) Release/obj.target/mysys_ssl/deps/libmariadbclient/mysys_ssl/my_aes.o
  CXX(target) Release/obj.target/mysys_ssl/deps/libmariadbclient/mysys_ssl/my_md5.o
  CXX(target) Release/obj.target/mysys_ssl/deps/libmariadbclient/mysys_ssl/my_rnd.o
  CXX(target) Release/obj.target/mysys_ssl/deps/libmariadbclient/mysys_ssl/my_sha1.o
  CXX(target) Release/obj.target/mysys_ssl/deps/libmariadbclient/mysys_ssl/my_sha2.o
  AR(target) Release/obj.target/deps/libmariadbclient/mysys_ssl/mysys_ssl.a
  COPY Release/mysys_ssl.a
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/bchange.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/bmove_upp.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-big5.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-bin.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-cp932.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-czech.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-eucjpms.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-euc_kr.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-extra.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-gb2312.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-gbk.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-latin1.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-mb.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-simple.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-sjis.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-tis620.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-uca.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-ucs2.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-ujis.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-utf8.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype-win1250ch.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/ctype.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/dtoa.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/int2str.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/is_prefix.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/llstr.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/longlong2str.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/my_strtoll10.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/my_vsnprintf.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/str2int.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strcend.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strcont.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strend.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strfill.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strmake.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strmov.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strmov_overlapp.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strnmov.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strxmov.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strxnmov.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/str_alloc.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/xml.o
  CC(target) Release/obj.target/strings/deps/libmariadbclient/strings/strnlen.o
  AR(target) Release/obj.target/deps/libmariadbclient/strings/strings.a
  COPY Release/strings.a
  CC(target) Release/obj.target/vio/deps/libmariadbclient/vio/vio.o
../deps/libmariadbclient/vio/vio.c: In function ‘vio_end’:
../deps/libmariadbclient/vio/vio.c:389:3: warning: implicit declaration of function ‘yaSSL_CleanUp’ [-Wimplicit-function-declaration]
   yaSSL_CleanUp();
   ^
  CC(target) Release/obj.target/vio/deps/libmariadbclient/vio/viopipe.o
  CC(target) Release/obj.target/vio/deps/libmariadbclient/vio/vioshm.o
  CC(target) Release/obj.target/vio/deps/libmariadbclient/vio/viosocket.o
  CC(target) Release/obj.target/vio/deps/libmariadbclient/vio/viossl.o
  CC(target) Release/obj.target/vio/deps/libmariadbclient/vio/viosslfactories.o
  AR(target) Release/obj.target/deps/libmariadbclient/vio/vio.a
  COPY Release/vio.a
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/adler32.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/compress.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/crc32.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/deflate.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/gzclose.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/gzlib.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/gzread.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/gzwrite.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/infback.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/inffast.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/inflate.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/inftrees.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/trees.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/uncompr.o
  CC(target) Release/obj.target/zlib/deps/libmariadbclient/zlib/zutil.o
  AR(target) Release/obj.target/deps/libmariadbclient/zlib/zlib.a
  COPY Release/zlib.a
  CXX(target) Release/obj.target/sqlclient/src/binding.o
../src/binding.cc: In member function ‘void Client::on_resultend()’:
../src/binding.cc:1170:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (r <= 0 || r >= sizeof(u64_buf))
                                        ^
../src/binding.cc:1179:42: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (r <= 0 || r >= sizeof(u64_buf))
                                          ^
../src/binding.cc:1186:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (r <= 0 || r >= sizeof(u64_buf))
                                        ^
../src/binding.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE Client::LastInsertId(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/binding.cc:1481:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (r <= 0 || r >= sizeof(u64_buf))
                                        ^
  SOLINK_MODULE(target) Release/obj.target/sqlclient.node
  COPY Release/sqlclient.node
make: Leaving directory `/home/vagrant/node-mariasql/build'
lru-cache@2.7.0 node_modules/lru-cache

nan@2.1.0 node_modules/nan
vagrant@vagrant-ubuntu-trusty-64:~/node-mariasql$ npm test

> mariasql@0.2.1 test /home/vagrant/node-mariasql
> node test/test.js

Executing "Client::version()"
Finished "Client::version()"
Executing "Client::escape()"
Finished "Client::escape()"
Executing "prepare()"
Finished "prepare()"
Executing "Non-empty threadId"
node: ../src/binding.cc:924: static void Client::cb_poll(uv_poll_t*, int, int): Assertion `status == 0' failed.
Aborted (core dumped)
npm ERR! Test failed.  See above for more details.
vagrant@vagrant-ubuntu-trusty-64:~/node-mariasql$ sudo service mysql start
mysql start/running, process 13096
vagrant@vagrant-ubuntu-trusty-64:~/node-mariasql$ npm test

> mariasql@0.2.1 test /home/vagrant/node-mariasql
> node test/test.js

Executing "Client::version()"
Finished "Client::version()"
Executing "Client::escape()"
Finished "Client::escape()"
Executing "prepare()"
Finished "prepare()"
Executing "Non-empty threadId"
Finished "Non-empty threadId"
Executing "Empty threadId (explicit disable)"
Finished "Empty threadId (explicit disable)"
Executing "escape()"
Finished "escape()"
Executing "isMariaDB()"
Finished "isMariaDB()"
Executing "serverVersion()"
Finished "serverVersion()"
Executing "Buffered result (defaults)"
Finished "Buffered result (defaults)"
Executing "Buffered result (useArray)"
Finished "Buffered result (useArray)"
Executing "Buffered result (metadata)"
Finished "Buffered result (metadata)"
Executing "Streamed result (defaults)"
Finished "Streamed result (defaults)"
Executing "Streamed result (useArray)"
Finished "Streamed result (useArray)"
Executing "Streamed result (metadata)"
Finished "Streamed result (metadata)"
Executing "Streamed result (INSERT)"
Finished "Streamed result (INSERT)"
Executing "Streamed result (START TRANSACTION, no data listener)"
Finished "Streamed result (START TRANSACTION, no data listener)"
Executing "lastInsertId()"
Finished "lastInsertId()"
Executing "multiStatements"
Finished "multiStatements"
Executing "Process queue before connection close"
Finished "Process queue before connection close"
Executing "Abort long running query"
Finished "Abort long running query"
Executing "Abort connection"
Finished "Abort connection"
Executing "clean up pending queries on close"
Finished "clean up pending queries on close"
Executing "keep pending queries on close"
Finished "keep pending queries on close"
vagrant@vagrant-ubuntu-trusty-64:~/node-mariasql$
```

**Comment by blah238** [comment]
Seems similar to https://github.com/mscdex/node-mariasql/issues/22

**Comment by blah238** [comment]
For reference here is the line in question: https://github.com/mscdex/node-mariasql/blob/1dd932281d7800f1fb0100883e43f64fc1c06f56/src/binding.cc#L924

**Comment by mscdex** [comment]
@blah238 A better version of the original fix has now been landed in f77b0c4de and 9cc00486d3. Let me know if it works for you too.

**Comment by blah238** [comment]
@mscdex Yep that looks like it's working, thanks!
